core/src/main/java/org/bitcoinj/utils/MonetaryFormat.java

```diff
@@ -233,15 +234,26 @@ public MonetaryFormat noCode() {
      */
     public MonetaryFormat code(int codeShift, String code) {
         checkArgument(codeShift >= 0);
-        Map<Integer, String> codes = new HashMap<Integer, String>();
-        if (this.codes != null)
-            codes.putAll(this.codes);
-        codes.put(codeShift, code);
+        final String[] codes = Arrays.copyOf(this.codes, this.codes.length);
+
+        codes[codeShift] = code;
         return new MonetaryFormat(negativeSign, positiveSign, zeroDigit, decimalMark, minDecimals, decimalGroups,
                 shift, roundingMode, codes, codeSeparator, codePrefixed);
     }
 
     /**
+     * Configure all currency codes as a batch.
+     * 
+     * @param codes
+     *            array of currency code representing the code at each shift
+     * from 0-8 decimal places. Use null where no specific code applies.
+     */
+    public MonetaryFormat codes(String[] codes) {
+        return new MonetaryFormat(negativeSign, positiveSign, zeroDigit, decimalMark, minDecimals, decimalGroups,
+                shift, roundingMode, Arrays.copyOfRange(codes, 0, MAX_DECIMALS), codeSeparator, codePrefixed);
```

Comment:
I think if codes.length > MAX_DECIMALS it should warn the caller, rather than silently capping the array.